internal/controlplane/handlers_artifacts.go

```diff
@@ -121,6 +126,12 @@ func (s *Server) GetArtifactByName(ctx context.Context, in *pb.GetArtifactByName
 		return nil, status.Errorf(codes.Unknown, "failed to get artifact versions: %s", err)
 	}
 
+	// Telemetry logging
+	logger.BusinessRecord(ctx).Provider = in.GetContext().GetProvider()
```

Comment:
Are there any calls that don't operate with `EntityContext`? Could we have a utility function that populates the telemetry using values read from the EntityContext that's itself read from the ctx?

Reply:
I was talking about this with @evankanderson. Essentially I was wondering if there's a better approach to declutter and limit the repetitions we do in the RPC handlers. Should there be a wrapper similar to what @JAORMX did for the CLIs? Or would it make sense to move context-related values like project and provider to context.Context and process/store them from the logging interceptor. I don't have a clear view of what's the best approach but it feels like there should be one.

Reply:
So since many calls use `authAndContextValidation` (which is confusingly define in handlers_profile.go but used in more handlers), would it be too sneaky to also get the telemetry store from the context there and log at least the project and provider? Logging the actual entries in a generic way might be tricky as the requests and replies are different for each request...

Reply:
I thought I had replied to this one. 

There's no unified approach to fetching the provider/project pair so far so I defaulted to explicitly call this on each handler instead of embedding it in an existing function (i.e. `authAndContextValidation`) that is frequently used but not common across all RPC handlers (i.e. `DeleteRepositoryByName`).

My suggestion is aligned with what you say - figure out a way to have a standard way of fetching the provider/project pair for all RPCs, but leave the other fields explicitly called from the handlers since they are custom to their request/response.

Reply:
Yes!  This is why we don't want writing TelemetryStore to be a "one and done" from a central place, but we should use this as pressure towards having a common way to get common fields (i.e. doing some code cleanup where it makes sense).